staging/src/k8s.io/cloud-provider/controllers/node/node_controller_test.go

```diff
@@ -162,6 +162,121 @@ func TestEnsureNodeExistsByProviderID(t *testing.T) {
 	}
 }
 
+func Test_ensureNodeProvidedIPExists(t *testing.T) {
+
+	testCases := []struct {
+		testName              string
+		node                  *v1.Node
+		nodeAddresses         []v1.NodeAddress
+		expectedNodeAddresses []v1.NodeAddress
+	}{
+		{
+			testName: "node provided IP exists",
+			node: &v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "node0",
+					Annotations: map[string]string{
+						cloudproviderapi.AnnotationAlphaProvidedIPAddr: "192.168.25.1",
+					},
+				},
+			},
+			nodeAddresses: []v1.NodeAddress{
+				{
+					Type:    v1.NodeInternalIP,
+					Address: "192.168.17.8",
+				},
+				{
+					Type:    v1.NodeInternalIP,
+					Address: "192.168.25.1",
+				},
```

Comment:
This test case only swaps two items, could we add a couple more so that we can see it successfully moving a few addresses around?

Reply:
Yep. Lets have the major cases where node IP is:

* First
* Middle
* Last